src/signalingclient/DefaultSignalingClient.ts

```diff
@@ -211,6 +214,25 @@ export default class DefaultSignalingClient implements SignalingClient {
       this.sendEvent(
         new SignalingClientEvent(this, SignalingClientEventType.WebSocketClosing, null)
       );
+
+      // Continue resetting the connection even if SDK does not receive the "close" event.
+      const scheduler: TimeoutScheduler = new TimeoutScheduler(
+        DefaultSignalingClient.CLOSE_EVENT_TIMEOUT_MS
+      );
+      const handler = (event: CloseEvent): void => {
+        this.webSocket?.removeEventListener('close', handler);
+        scheduler.stop();
+        this.closeEventHandler(event);
+      };
+
+      // Remove the existing close handler to prevent SDK from opening a new connection.
+      this.webSocket?.removeEventListener('close', this.closeEventHandler);
```

Comment:
Do we need to check for undefined/null webSocket here (as the check above in the if condition seems to assume it exists and same as the add below).

Reply:
oops. Not `this.webSocket?` but I should've checked the existence of `removeEventListener`.

Reply:
I'll add the `/* istanbul ignore next */` since there's no point of testing a WebSocket object without `removeEventListener`. Builders cannot pass the custom WebSocket object.

Reply:
We should just mark these classes internal in next major version :)